src/main/java/com/google/research/bleth/servlets/EnqueueExperimentServlet.java

```diff
@@ -0,0 +1,233 @@
+                "was created and written to db.");
+
+        int legalConfigurationsCount = 0;
+        for (List<PropertyWrapper> configuration : configurations) {
+            if (validateArguments(configuration)) {
+                AppEngineHttpRequest httpRequest = toHttpRequest(configuration, beaconsNum,
+                        experimentId, experimentTitle);
+                log.info("A new AppEngineHttpRequest was created: "
+                + httpRequest.toString());
+                enqueueTask(httpRequest);
+                log.info("A new Task was created and enqueued.");
+                legalConfigurationsCount++;
+            }
+        }
+
+        try {
+            experiment = datastore.get(experimentId);
+            experiment.setProperty(Schema.Experiment.simulationsLeft, legalConfigurationsCount);
+            datastore.put(experiment);
+            log.info("Experiment entity with id " + KeyFactory.keyToString(experimentId) +
+                    "was updated with simulationsLeft=" + legalConfigurationsCount);
+        } catch (EntityNotFoundException e) {
+            response.setContentType("text/plain;");
+            response.getWriter().println(e.getMessage());
+        }
+
+        response.setContentType("text/plain;");
+        response.getWriter().println(legalConfigurationsCount + " tasks have been added to queue.");
+    }
+
+    @AutoValue
+    public static abstract class PropertyWrapper {
+        public static PropertyWrapper create(String property, Number value) {
+            return new AutoValue_EnqueueExperimentServlet_PropertyWrapper(property, value);
+        }
+        public abstract String property();
+        public abstract Number value();
+    }
+
+    private Set<List<PropertyWrapper>> createConfigurations(HttpServletRequest request) {
+        Set<PropertyWrapper> roundsNumValues = createIntValues(request, "roundsNum");
+        Set<PropertyWrapper> rowsNumValues = createIntValues(request, "rowsNum");
+        Set<PropertyWrapper> colsNumValues = createIntValues(request, "colsNum");
+        Set<PropertyWrapper> observersNumValues = createIntValues(request, "observersNum");
+        Set<PropertyWrapper> awakenessCycleValues = createIntValues(request, "awakenessCycle");
+        Set<PropertyWrapper> awakenessDurationValues = createIntValues(request, "awakenessDuration");
+        Set<PropertyWrapper> transmissionThresholdRadiusValues = createDoubleValues(request, "transmissionThresholdRadius");
+
+        return Sets.cartesianProduct(
+                roundsNumValues,
+                rowsNumValues,
+                colsNumValues,
+                observersNumValues,
+                awakenessCycleValues,
+                awakenessDurationValues,
+                transmissionThresholdRadiusValues
+        );
+    }
+
+    private Set<PropertyWrapper> createIntValues(HttpServletRequest request, String parameter) {
```

Comment:
Is this function necessary? It seems like createDoubleValues is sufficient.

Reply:
I just checked and it seems like it does required: e.g Double 10 will be converted into the string '10.0' and then the worker servlet won't be able to parse it using `Integer.parseInt()`.

Reply:
I assume Double.parseDouble(...).intValue() can fix this, but I'm not sure if it's better than the current situation...